huaweicloud/services/cfw/resource_huaweicloud_cfw_capture.go

```diff
@@ -0,0 +1,498 @@
+	raw := rawArray[0].(map[string]interface{})
+	return map[string]interface{}{
+		"dest_port":   utils.ValueIgnoreEmpty(raw["dest_port"]),
+		"protocol":    raw["protocol"],
+		"source_port": utils.ValueIgnoreEmpty(raw["source_port"]),
+	}
+}
+
+func resourceCaptureRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	client, err := conf.NewServiceClient("cfw", region)
+	if err != nil {
+		return diag.Errorf("error creating CFW client: %s", err)
+	}
+
+	name := d.Get("name").(string)
+	instanceID := d.Get("fw_instance_id").(string)
+	capture, err := GetCapture(client, name, instanceID)
+	if err != nil {
+		return common.CheckDeletedDiag(d, parseError(err), "error retrieving capture")
+	}
+
+	id := utils.PathSearch("task_id", capture, "")
+	d.SetId(id.(string))
+
+	createdDateStr := utils.PathSearch("created_date", capture, "").(string)
+	createdTimestamp := utils.ConvertTimeStrToNanoTimestamp(createdDateStr, "2006/01/02 15:04:05") / 1000
+	modifiedDateStr := utils.PathSearch("modified_date", capture, "").(string)
+	modifiedTimestamp := utils.ConvertTimeStrToNanoTimestamp(modifiedDateStr, "2006/01/02 15:04:05") / 1000
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("name", capture, nil)),
+		d.Set("destination", flattenGetDestinationAddressDto(capture)),
+		d.Set("source", flattenGetSourceAddressDto(capture)),
+		d.Set("service", flattenGetServiceDto(capture)),
+		d.Set("duration", utils.PathSearch("duration", capture, nil)),
+		d.Set("max_packets", utils.PathSearch("max_packets", capture, nil)),
+		d.Set("status", utils.PathSearch("status", capture, nil)),
+		d.Set("created_at", utils.FormatTimeStampRFC3339(createdTimestamp, false)),
+		d.Set("updated_at", utils.FormatTimeStampRFC3339(modifiedTimestamp, false)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func GetCapture(client *golangsdk.ServiceClient, name, instanceID string) (interface{}, error) {
+	httpUrl := "v1/{project_id}/capture-task"
+	path := client.Endpoint + httpUrl
+	path = strings.ReplaceAll(path, "{project_id}", client.ProjectID)
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	offset := 0
+	for {
+		path := fmt.Sprintf("%s?fw_instance_id=%s&limit=100&offset=%d", path, instanceID, offset)
+		resp, err := client.Request("GET", path, &opt)
+
+		if err != nil {
+			return nil, err
+		}
+
+		respBody, err := utils.FlattenResponse(resp)
+		if err != nil {
+			return nil, err
+		}
+
+		curJson := utils.PathSearch("data.records", respBody, make([]interface{}, 0))
+		curArray := curJson.([]interface{})
+
+		if len(curArray) == 0 {
+			return nil, golangsdk.ErrDefault404{}
+		}
+
+		findCaptureStr := fmt.Sprintf("[?name=='%s']|[0]", name)
+		capture := utils.PathSearch(findCaptureStr, curJson, nil)
+		if capture != nil {
+			return capture, nil
+		}
+
+		offset += 100
```

Comment:
the response has `total`, it's no need to search the next page if the datas are less than 100.